src/app-layer-detect-proto.c

```diff
@@ -345,7 +345,9 @@ static AppProto AppLayerProtoDetectPPGetProto(Flow *f,
             SCLogDebug("toserver - Probing parser found for source port %"PRIu16, f->sp);
 
             /* found based on source port, so use sp registration */
-            pe2 = pp_port_sp->sp;
+            pe2 = pp_port_sp->dp;
+            StreamTcpPacketSwitchDir(NULL, p);
```

Comment:
this use of StreamTcpPacketSwitchDir doesn't make sense. It flips the direction flags for this one packet, but does nothing for the rest of the flow.